src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -76,6 +75,24 @@ internal class TracerProviderSdk : TracerProvider
                 {
                     OpenTelemetrySdkEventSource.Log.ActivityStarted(activity);
 
+                    if (string.IsNullOrEmpty(activity.Source.Name))
```

Comment:
we can have a list of Activity names, for which we need to run samplers, and only run this for those opt-in activity names. Running this for *all* the legacy activities would affect hot path performance.
(even with list of opt-in names, we need to run benchmarks to see if there is significant regression)

Reply:
Something like the below:

List<string> LegacyActivityNames; which is populated when instrumentation calls `AddLegacyActivity(string name)`.

if this list is empty, then bypass the whole section. This would be the case when someone is using pure ActivitySource based activities only.

Reply:
I have added a HashSet<string> to store all the operation names for the DiagnosticSource instrumentations that were added.